File: Problems/Algorithms/1926. Nearest Exit from Entrance in Maze/nearest_exit.go
func nearestExit(maze [][]byte, entrance []int) int {
    n, m := len(maze), len(maze[0])

    neighbors := [4][2]int{{0, 1}, {0, -1}, {1, 0}, {-1, 0}}
    visited := make([][]bool, n)
    for row := range visited {
        visited[row] = make([]bool, m)
    }

    queue := make([][3]int, 0)
    sx, sy := entrance[0], entrance[1]
    queue = append(queue, [3]int{sx, sy, 0})
    visited[sx][sy] = true

    for len(queue) > 0 {
        r, c, d := queue[0][0], queue[0][1], queue[0][2]
        queue = queue[1:]

        if d != 0 && (r == 0 || r == n-1 || c == 0 || c == m-1) {
            return d
        }

        for _, neighbor := range neighbors {
            x, y := r + neighbor[0], c + neighbor[1]
            if x >= 0 && y >= 0 && x < n && y < m && !visited[x][y] && maze[x][y] == '.' {
                visited[x][y] = true
                queue = append(queue, [3]int{x, y, d+1})
            }
        }
    }

    return -1
}

File: Problems/Algorithms/1926. Nearest Exit from Entrance in Maze/nearest_exit.py
#!/usr/bin/env python3
# coding: utf-8

from typing import List


class Solution:
    def nearestExit(self, maze: List[List[str]], entrance: List[int]) -> int:
        n, m = len(maze), len(maze[0])

        neighbors = [(0, 1), (0, -1), (1, 0), (-1, 0)]
        visited = set()

        queue = [tuple(entrance)]
        visited.add(tuple(entrance))
        steps = 0
        while queue:
            steps += 1
            for _ in range(len(queue)):
                x, y = queue.pop(0)
                for dx, dy in neighbors:
                    nx, ny = x + dx, y + dy
                    if 0 <= nx < n and 0 <= ny < m and maze[nx][ny] == '.' and (nx, ny) not in visited:
                        if nx == 0 or nx == n - 1 or ny == 0 or ny == m - 1:
                            return steps
                        visited.add((nx, ny))
                        queue.append((nx, ny))

        return -1
